### ISSUE TITLE
'SAME' and 'VALID' padding support for Conv2d and related ops

### ISSUE THREAD

**Issue by AjinkyaPrabhu** [pytorch/pytorch#17953]
## 🚀 Feature
Support for Same and valid convolutions in the Conv2d and related operations

## Motivation

I just started using pytorch (was using tensorflow before) and really like the pythonic API, only problem I encounter is when creating a model , I have recalculate the parameters for all the layers.

## Pitch

adding padding='same' and padding='valid' in the Conv2d constructor,and to be used in tandem with a `torch.compile(model=model,input_shape=(3,224,224))`. and throw an error if a forward pass is attempted if the `padding=='same'` and t`orch.compile()` is not run.

## Alternatives

recalculating padding for every forward pass ,but that wouldn't be very efficient

**Comment by RicCu** [comment]
Hi,
This feature has been proposed for the 1.1 milestone. You can find the discussion at #3867

**Comment by soumith** [comment]
closing as duplicate of #3867
